### DISCUSSION TITLE
THREDDS (netCDF-Java) roadmap

### DISCUSSION THREAD

**Discussion by haileyajohnson** [Unidata/netcdf-java#905] [General]
Hello THREDDS users - below, please find the current develop "roadmap" for the netCDF-Java project. You can also find the TDS roadmap [here](https://github.com/Unidata/tds/discussions/172).
_Note_ that no plans are final, development is always changing, and we welcome discussion, feedback, and contributions!

### Version 5.x
The current stable version of netCDF-Java is _**5.4**_. We anticipate that _**5.x**_ will exist as a long term release and continue to support the TDS. Several features in development are slated for inclusion in future _**5.x**_ releases, notably Zarr/NCZarr support and RPC communication, as well as continued work on object storage.

### Version 6.x and 7.x
_**NetCDF 6-BETA**_ was released a few months ago. The move from _**5.x**_ to _**6.x**_ involves significant API changes, as v6 is the first version of netCDF-Java that implements strict immutability. As such, we consider v6 a short-term transition release, and do not recommend 6.x in production, however beta testers and feedback are encouraged! Similarly, version _**7**_ is another transitional release on the way to the newer API (see below).

### Version 8.x
**_8.x_** will be the first version of netCDF-Java to fully implement the new API. Its primary goal is to be a modernization of the netCDF-Java library; a lighter-weight version with a strict commitment to immutability. It will likely not include all of the features and IOSPs found in _**5.x**_, since with version 8, we aim to support only modern, clean code. We will continue to port features and IOSP to future versions of _**8.x**_ and up. As always, contributions welcome!

**Comment by mcucchi9** [comment]
Hi @haileyajohnson,

is there a planned (approximate) date for the release of the next stable version of netCDF-Java?

I'm particularly interested in the implementation of the fix to the issue discussed at https://github.com/Unidata/netcdf-java/issues/997, which has been merged in `maint-5.x` (https://github.com/Unidata/netcdf-java/commit/a3f554ff7ddbe9c6944c3d55faab592d51e0351d).

Thanks!

**Reply by haileyajohnson** [reply]
We hope to have 5.5.3 released within the next two weeks. At the moment, we're only holding off on a release to make sure than any bug fixes in TDS 5.4 propagate down into the netcdf-java release as well. If you need that fix more immediately, you can grab the latest snapshot release [here](https://artifacts.unidata.ucar.edu/repository/downloads-netcdf-java/5.5.3/netcdfAll-5.5.3-SNAPSHOT.jar).
